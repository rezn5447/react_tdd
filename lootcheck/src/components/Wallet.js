import React, { PropTypes, Component } from 'react';
import { connect } from 'react-redux';

export class Wallet extends Component {
  render() {
    return (
      <div>
        <h3 className="balance">Wallet Balance: {this.props.balance}</h3>
      </div>
    );
  }
}

const mapStateToProps = state => {
  const { balance } = this.state;
  return balance;
};

export default connect(mapStateToProps, null)(Wallet);
