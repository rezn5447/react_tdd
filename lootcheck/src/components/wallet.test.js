import React from 'react';
import { shallow } from 'enzyme';
import { Wallet } from './Wallet';

describe('Wallet', () => {
  const props = { balance: 20 };
  const wallet = shallow(<Wallet {...props} />);

  it('should render correctly', () => {
    expect(wallet).toMatchSnapshot();
  });

  it('should display the balance from props', () => {
    expect(wallet.find('.balance').text()).toEqual('Wallet Balance: 20');
  });

  it('should have an input to deposit or withdraw from the balance', () => {
    expect(wallet.find('.input-wallet').exists()).toBe(true);
  });

  it('should have a button for withdraw', () => {
    expect(wallet.find('.btn-withdraw').exists()).toBe(true);
  });

  it('should have a button for deposit', () => {
    expect(wallet.find('.btn-deposit').exists()).toBe(true);
  });

  describe('when the user types into the wallet Input', () => {
    const userBalance = '25';
    beforeEach(() => {
      wallet
        .find('.input-wallet')
        .simulate('change', { target: { value: userBalance } });
    });

    it('should update the local wallet balance in `state` and converts it to a number', () => {
      expect(wallet.state().balance).toEqual(parseInt(userBalance, 10));
    });
  });
});
