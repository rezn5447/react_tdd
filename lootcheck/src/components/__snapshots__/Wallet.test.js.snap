// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    balance={20}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="balance"
>
          Wallet Balance: 
          20
</h3>,
        <br />,
        <Row>
          <Input
                    checked={false}
                    className="input-wallet"
                    label="Amount"
                    onChange={[Function]}
                    s={4}
                    type="text"
          />
          <Button
                    className="btn-deposit"
                    node="button"
                    onClick={undefined}
                    waves="light"
          >
                    Deposit
          </Button>
          <Button
                    className="btn-withdraw"
                    node="button"
                    onClick={undefined}
                    style={
                              Object {
                                        "margin": "0 10px",
                                      }
                    }
                    waves="light"
          >
                    Withdraw
          </Button>
</Row>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Wallet Balance: ",
            20,
          ],
          "className": "balance",
        },
        "ref": null,
        "rendered": Array [
          "Wallet Balance: ",
          20,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Input
              checked={false}
              className="input-wallet"
              label="Amount"
              onChange={[Function]}
              s={4}
              type="text"
/>,
            <Button
              className="btn-deposit"
              node="button"
              onClick={undefined}
              waves="light"
>
              Deposit
</Button>,
            <Button
              className="btn-withdraw"
              node="button"
              onClick={undefined}
              style={
                            Object {
                                          "margin": "0 10px",
                                        }
              }
              waves="light"
>
              Withdraw
</Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "className": "input-wallet",
              "label": "Amount",
              "onChange": [Function],
              "s": 4,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Deposit",
              "className": "btn-deposit",
              "node": "button",
              "onClick": undefined,
              "waves": "light",
            },
            "ref": null,
            "rendered": "Deposit",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Withdraw",
              "className": "btn-withdraw",
              "node": "button",
              "onClick": undefined,
              "style": Object {
                "margin": "0 10px",
              },
              "waves": "light",
            },
            "ref": null,
            "rendered": "Withdraw",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="balance"
>
            Wallet Balance: 
            20
</h3>,
          <br />,
          <Row>
            <Input
                        checked={false}
                        className="input-wallet"
                        label="Amount"
                        onChange={[Function]}
                        s={4}
                        type="text"
            />
            <Button
                        className="btn-deposit"
                        node="button"
                        onClick={undefined}
                        waves="light"
            >
                        Deposit
            </Button>
            <Button
                        className="btn-withdraw"
                        node="button"
                        onClick={undefined}
                        style={
                                    Object {
                                                "margin": "0 10px",
                                              }
                        }
                        waves="light"
            >
                        Withdraw
            </Button>
</Row>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Wallet Balance: ",
              20,
            ],
            "className": "balance",
          },
          "ref": null,
          "rendered": Array [
            "Wallet Balance: ",
            20,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Input
                checked={false}
                className="input-wallet"
                label="Amount"
                onChange={[Function]}
                s={4}
                type="text"
/>,
              <Button
                className="btn-deposit"
                node="button"
                onClick={undefined}
                waves="light"
>
                Deposit
</Button>,
              <Button
                className="btn-withdraw"
                node="button"
                onClick={undefined}
                style={
                                Object {
                                                "margin": "0 10px",
                                              }
                }
                waves="light"
>
                Withdraw
</Button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "checked": false,
                "className": "input-wallet",
                "label": "Amount",
                "onChange": [Function],
                "s": 4,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Deposit",
                "className": "btn-deposit",
                "node": "button",
                "onClick": undefined,
                "waves": "light",
              },
              "ref": null,
              "rendered": "Deposit",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Withdraw",
                "className": "btn-withdraw",
                "node": "button",
                "onClick": undefined,
                "style": Object {
                  "margin": "0 10px",
                },
                "waves": "light",
              },
              "ref": null,
              "rendered": "Withdraw",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
